src/GafferScene/RenderController.cpp

```diff
@@ -95,6 +97,61 @@ bool cameraGlobalsChanged( const CompoundObject *globals, const CompoundObject *
 	return *camera1 != *camera2;
 }
 
+// This is for the very specific case of determining change for global
+// attributes, where we need to avoid comparisons of certain synthetic members
+// that are only present in previousFullAttributes.
+bool globalAttributesChanged( const CompoundObject* globalAttributes, const CompoundObject* previousFullAttributes )
+{
+	static std::set<IECore::InternedString> ignoredMembers = { IECore::InternedString( "sets" ) };
```

Comment:
Went back and forwards on just hard-coding `if( it1->first != staticSetsString )` and the offsets, but it seemed more fragile, given the use in the early-out clauses, etc... Happy to do that instead if preferred.

Reply:
"Why you shouldn't use set (and what you should use instead)" is a decent read : http://lafstern.org/matt/col1.pdf. The relevant bits here are that our number of elements is small and we build the container once at startup and never need to edit it again, hence `set` has overheads we pay for but don't benefit from. I agree the code maintenance is a little easier with `ignoredMembers` though, so suggest using `const boost::flat_set`instead (note the `const`).

Reply:
Ah, good to know, thanks for the link. https://github.com/GafferHQ/gaffer/pull/3606/commits/8e9bcd830907ce70768a570f0b9e2a4c97cdfc9b.